### ISSUE TITLE
getInheritedClasses shows wrong results when parent folder is not expanded

### ISSUE THREAD

**Issue by abuntrock** [OpenModelica/OpenModelica#12902]
### Description

For some classes the CLI call `getInhertitedClasses` depends on the opening status of its parent package. I was not able to build a minimal example yet, but you should have acess to the BRSL 3.0.0 via @niklwors 

### Steps to Reproduce

1. Load the BRSL3.0.0 and call `getInheritedClasses(Rexroth_BRSL.Rexroth_MobileControls.ControlBlocks.RS12.RS12Q)` the result is empty.
2. Do the same call again. The result is:
`{Rexroth_BRSL.Generic.Auxiliaries.UsageStatistics.BaseUsageStatistic,Rexroth_BRSL.Rexroth_MobileControls.ControlBlocks.RS12.RS12Q.RS12,Rexroth_BRSL.Rexroth_MobileControls.ControlBlocks.RS12.RS12Q.RS1x_SpringSetup_Interface}` which is wrong.
When you expand the BRSL package afterwards the follwoing error is displayed 
![{E07EE469-F283-4CB7-93DA-2E84EEFE94FE}](https://github.com/user-attachments/assets/8c755c31-1424-4896-b4db-a28f2925d57d)
When opening the class itself, the following error is displayed 
![{162D2C93-5D15-48BA-978D-48646D454B17}](https://github.com/user-attachments/assets/82232506-fb8b-4fc8-828d-30abb1165f42)
However, source control does not show any changes to the file and the model works completely fine when not calling `getInhertitedClasses`
4. Unload the BRSL, reload it again and open all parent folders of `Rexroth_BRSL.Rexroth_MobileControls.ControlBlocks.RS12.RS12Q`. Then call `getInheritedClasses(Rexroth_BRSL.Rexroth_MobileControls.ControlBlocks.RS12.RS12Q)` again. Now the result is 
`{Rexroth_BRSL.Generic.Auxiliaries.UsageStatistics.BaseUsageStatistic,Rexroth_BRSL.Rexroth_MobileControls.ControlBlocks.RS12.Internal.RS12,Rexroth_BRSL.Rexroth_MobileControls.ControlBlocks.Interfaces.RS1x_SpringSetup_Interface}` which is correct.

In the incorrect result the path to the input class is somehow used for the inherited classes (except for the first one which is correct there as well) and then just the name of the inherited class is appended at the end. Maybe this helps when trying to find the problem.

### Expected Behavior

The list of inheritedClasses is always correct no matter if the first parent package of the class is openend when calling `getInhertitedClasses`.

### Version and OS

 - OpenModelica Version: v1.24.0-dev.beta.0 
 - OS: Windows 11, 64 bit
 - MSL 4.0.0, BRSL 3.0.0

Thanks for looking into this
Aaron Buntrock - Bosch Rexroth

**Comment by perost** [comment]
MWE: 
```Modelica
loadString("
  package P1
    package P2
      package P3
        extends P1.P2.Icons.Package;

        model M
          extends P1.P2.P3.Internal.M;
        end M;

        package Internal
          model M
          end M;
        end Internal;
      end P3;

      package Icons
        package Package
        end Package;
      end Icons;
    end P2;
  end P1;
");

getInheritedClasses(P1.P2.P3.M);
getErrorString();
getInheritedClasses(P1.P2.P3.M);
getErrorString();
```
gives:
```
true
{}
"[<interactive>:8:11-8:38:writable] Error: Base class P1.P2.P3.Internal.M not found in scope M.
"
{P1.P2.P3.M.M}
""
```
I assume it has something to do with the shortcuts we take when instantiating packages for `getInheritedClasses`, but I'll have to take a closer look at what's going wrong here.

**Comment by abuntrock** [comment]
Thanks for finding a minimum working example for this. I was worried that it has something to do with our replaceable usage in our BRSL example but this does not seem to be the case.

**Comment by abuntrock** [comment]
Works perfect with today's nightly. Thanks a lot.

**Comment by abuntrock** [comment]
@casella If there is the need to build an OMEdit 1.24.1 release we would be glad if this ticket could make it inside.

**Comment by perost** [comment]
> @casella If there is the need to build an OMEdit 1.24.1 release we would be glad if this ticket could make it inside since this was a regression problem that worked for example in 1.22.0

I'm not sure what the plan is for 1.24.1, but I ported the fix to the 1.24 maintenance branch in #13023 so it'll be included if and when we make such a release.

**Comment by casella** [comment]
@abuntrock, we don't have a specific release date yet for 1.24.1, but it will likely be in November, so it collects enough critical fixes well ahead of 1.25.0. Of course if this is critical for you to get in an official release, we can do it right away, and add possible further fixes in a 1.24.2 release, just let us know.

**Comment by abuntrock** [comment]
November is good for us. This issue here just affects some backend stuff for us which we can still use an older OMEdit version for until 1.21.1 is out.
